src/cdk/constructs/CloudFrontDistribution.ts

```diff
@@ -55,40 +64,50 @@ export class CloudFrontDistribution extends Construct {
 
     const s3Origin = new origins.S3Origin(staticBucket)
 
-    this.cf = new cloudfront.Distribution(this, id, {
-      defaultBehavior: {
-        origin: s3Origin,
-        edgeLambdas: [
-          {
-            functionVersion: requestEdgeFunction.currentVersion,
-            eventType: cloudfront.LambdaEdgeEventType.ORIGIN_REQUEST
-          },
-          {
-            functionVersion: responseEdgeFunction.currentVersion,
-            eventType: cloudfront.LambdaEdgeEventType.ORIGIN_RESPONSE
-          }
-        ],
-        cachePolicy: splitCachePolicy
-      },
-      defaultRootObject: '',
-      additionalBehaviors: {
-        ['/_next/data/*']: {
+    if (customCloudFrontId && customCloudFrontDomainName) {
+      this.cf = cloudfront.Distribution.fromDistributionAttributes(this, id, {
```

Comment:
we also need to add new rules and lambda to existing cloudfront

Reply:
we can call `this.cf.addBehaviour`

Reply:
Unfortunately, this isn't supported by the cloudformation stack. we can only add existing id to the cloud formation template (which is what I did here by setting it to the output).
https://github.com/aws/aws-cdk/issues/12524

Reply:
hmm, I didn't see this comment https://github.com/DBB-Software/next-serverless-deployment/pull/33#discussion_r1719726020. let me check this out today.